packages/graphql/src/modules/user/app/authenticateFacebookUser.ts

```diff
@@ -39,66 +39,37 @@ const authenticateFacebookUser = ({
 
   const {id: fbID, email, name} = socialUserInfo;
 
-  const users = await userRepository.findUserByEmailOrProviderId({
+  const user = await userRepository.findUserByEmailOrProviderId({
     providersIds: [{id: fbID, providedBy: 'facebook'}],
     emails: [{email, providedBy: 'facebook'}],
   });
 
-  const existingUser = users && users[0];
+  if (user) {
+    let updatedUser = User.updateName(user, name);
 
-  if (existingUser) {
-    let updatedUser = existingUser;
-    // verify if name was updated
-    if (name.length !== existingUser.name.length) {
-      updatedUser = {...existingUser, name};
-      await userRepository.update(updatedUser);
-    }
-
-    const dbFbID = updatedUser.providersIds.find(id => id.providedBy === PROVIDERS.FACEBOOK);
-
-    if (dbFbID?.id !== fbID) {
-      const updatedProvidersIds = updatedUser.providersIds.map(providerId =>
-        providerId.providedBy === 'facebook' ? ({id: fbID, providedBy: 'facebook'} as ProvidersId) : providerId,
-      );
-
-      updatedUser = {...updatedUser, providersIds: updatedProvidersIds};
-
-      await userRepository.update(updatedUser);
-    }
-
-    const existingFbProfileImages = updatedUser.profileImages.find(
-      profileImage => profileImage.providedBy === 'facebook',
-    );
+    updatedUser = User.updateFacebookId(updatedUser, fbID);
```

Comment:
To be honest, I'm wondering about the need for these two functions `updateName` and `updateFacebookId`. Do we really need them? I mean, every time a user logs into we will reach line 58, even when he doesn't have any data to be updated. So, in order to make it simpler, I would create a `shouldUpdateUser` which receives a user instance in order to avoid a possible database hit. This function should check wheater the user name, facebookId or facebook profile image needs to be updated and return a boolean. 
Just make sure to keep the check and the routine to update facebook profile image since it hits an external provider.

Reply:
The problem may be the function's name. These two functions actually verify if the desired property needs to be changed

Reply:
I don't think so. Just create a function with a generic name like `shouldUpdateUser`. In the future, we might have more properties to be checked, then it's bad if we would need to create one function for each property in order to update the user.